### ISSUE TITLE
Download Symbols command fails when launch configuration saved to workspace file

### ISSUE THREAD

**Issue by vjekob** [microsoft/AL#6306]
**Describe the bug**
When a workspace file is present and launch configuration is stored in the workspace file (as opposed in the launch.json file in the .vscode subfolder), starting `AL: Download Symbols` fails to read the launch configuration from the workspace configuration, and behaves as if there is no launch configuration.

In addition, during the execution of `AL: Download Symbols` command, there will be an additional prompt to create a configuration, and when selecting a configuration from the list, the current configuration in the workspace file will be replaced with the new default configuration of the chosen type (Local or Microsoft Cloud).

**To Reproduce**
1. Run `AL: Go!` and create a new AL project
2. Delete the launch.json file from the .vscode subfolder of the new project
3. Click **File > Close Folder** to close the current folder
4. Click **File > Add Folder to Worspace** and then add the folder created in step 1
5. Click **File > Save Workspace As** and save the workspace as "Demo.code-workspace"
6. In the Activity Bar, click the Run icon
7. Click **create a launch.json file**
8. In the **Select a workspace folder** prompt, select **worspace**
9. Complete the configuration of the AL Language launch configuration
10. Run `AL: Download Symbols`

``` AL Code to reproduce the issue ```

**Expected behavior**
This is how it should be:
1. AL Language extension should be able to read the launch configuration from the workspace file
2. If launch configuration is present in the workspace file, the `AL: Download Symbols` command must not replace it

**5. Versions:**
 - AL Language: 6.1.362735
 - Business Central: 17

**Comment by darrenwaller** [comment]
If I create a launch.json file with nothing more than {} in the file then it does pickup the configuration from the workspace file.

**Comment by vjekob** [comment]
> If I create a launch.json file with nothing more than {} in the file then it does pickup the configuration from the workspace file.

This is a workaround, though, isn't it? It doesn't fix the problem, it puts it under the carpet.

Also - is this documented anywhere? How do people figure this out? (And why should they - it's still a bug, other extensions know how to handle this)

**Comment by kalberes** [comment]
Yes, this is a bug.  We have the configuration files hard coded for now in the .vscode folder. Don't expect a fast resolution though, since this affects a few (unnecessary before workspace era)  couplings.

**Comment by vjekob** [comment]
> Yes, this is a bug. We have the configuration files hard coded for now in the .vscode folder. Don't expect a fast resolution though, since this affects a few (unnecessary before workspace era) couplings.

Sure, I am glad that at least this is something you are aware of and that will be solved at a point. Thanks!

**Comment by DavidFeldhoff** [comment]
I just want to point out that we are also waiting for a fix :) Any news?

**Comment by SBalslev** [comment]
Hi! Thank you for taking the time to report this — we really appreciate the community's help in making Business Central better.

Since this issue was filed, we've shipped many updates to the AL developer tools and platform. Unfortunately, we weren't able to follow up on every older issue, and we're sorry about that.

If this is still a problem for you on a recent version of Business Central, please let us know by opening a new issue with updated repro steps — we'd love to take another look. Thank you for your patience and continued feedback!
